app/src/main/java/com/yml/womensafety/TipsForWomenSafetyAdapter.kt

```diff
@@ -0,0 +1,37 @@
+package com.yml.womensafety
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.yml.womensafety.navigationdrawer.SafetyTips
+
+class TipsForWomenSafetyAdapter(private val tipsTextList: ArrayList<SafetyTips>) :
+    RecyclerView.Adapter<TipsForWomenSafetyAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): TipsForWomenSafetyAdapter.ViewHolder {
+        val v =
+            LayoutInflater.from(parent.context).inflate(R.layout.women_safety_item, parent, false)
+        return ViewHolder(v)
+    }
+
+    override fun onBindViewHolder(holder: TipsForWomenSafetyAdapter.ViewHolder, position: Int) {
+        holder.bindItems(tipsTextList[position])
+    }
+
+    override fun getItemCount(): Int {
+        return tipsTextList.size
+    }
+
+    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
Can we have the name of this viewholder more readable and in the context?

Reply:
> Can we have the name of this viewholder more readable and in the context?

Done